mod.Lua

```diff
@@ -1,22 +1,20 @@
+local total_xp = 0 -- Create variables outside the hook, so we dont recreate them every hookcall
+local notification_text = ""
 
-local previous_total_xp = 0
-Hooks:PreHook(ExperienceManager, "mission_xp_award", "xp_notifier_before_xp_increase", function ()
-	previous_total_xp = get_current_total_xp()
-end)
+function xp_string(n)
+	if managers.experience ~= nil then -- Check for null/nil so we dont crash
+	   return managers.experience:experience_string(n)
```

Comment:
This is actually a good idea. I didn't think the `managers.hud:show_hint` call below could crash the game, but there's an edge case where it does and somebody opened a PR :)
This made me realize that defensive programming is necessary here because any error here crashes the game :P

Reply:
By the way, I think an early return would be better here